src/core/lispStream.cc

```diff
@@ -5152,11 +5100,94 @@ static int file_listen(T_sp stream, int fileno) {
     FEerror("Unsupported Windows file type: ~A", 1, make_fixnum(GetFileType(hnd)).raw_());
     break;
   }
+#else
+  /* Method 1: poll, see POLL(2)
+     Method 2: select, see SELECT(2)
+     Method 3: ioctl FIONREAD, see FILIO(4)
+     Method 4: read a byte. Use non-blocking I/O if poll or select were not
+               available. */
+  int result;
+#if defined(HAVE_POLL)
+  struct pollfd fd = {fileno, POLLIN, 0};
+restart_poll:
+  result = poll(&fd, 1, 0);
+  if (UNLIKELY(result < 0)) {
+    if (errno == EINTR)
+      goto restart_poll;
+    goto listen_error;
+  }
+  if (fd.revents == 0) {
+    return CLASP_LISTEN_NO_CHAR;
+  }
+  /* When read() returns a result without blocking, this can also be
+     EOF! (Example: Linux and pipes.) We therefore refrain from simply
+     doing  { return CLASP_LISTEN_AVAILABLE; }  and instead try methods
+     3 and 4. */
+#elif defined(HAVE_SELECT)
+  fd_set fds;
+  struct timeval tv = {0, 0};
+  /*
+   * Note that the following code is fragile. If the file is closed (/dev/null)
+   * then select() may return 1 (at least on OS X), so that we return a flag
+   * saying characters are available but will find none to read. See also the
+   * code in cl_clear_input().
+   */
```

Comment:
I don't understand this comment. It makes it sound like the code is broken on `/dev/null`. But it looks like if `select` returns 1 we continue on to `ioctl` or `read` anyway, rather than returning a flag saying characters are available.

Reply:
To be honest, I don't either. I didn't add it. It was already in Clasp. I think it might be obsolete.

Reply:
I just removed the comment.